SQL/sqlRules.json

```diff
@@ -0,0 +1,42 @@
+{
+  "SQL_Server_Auth_EntraIDIntegrated": {
+    "expected": true,
+    "explanation": "The server should be integrated with Azure Active Directory for authentication and authorization. For more information: https://learn.microsoft.com/en-us/sql/relational-databases/security/authentication-access/azure-ad-authentication-sql-server-overview?view=sql-server-ver16",
+    "function": "hasAADIntegrationEnabled"
+  },
+  "SQL_Server_Auditing" : {
+    "expected": true,
+    "explanation": "The server should have auditing enabled. For more information: https://learn.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-database-engine?view=sql-server-ver16",
+    "function": "hasAuditingEnabled"
+  },
+  "SQL_Server_TLS_1.2": {
+    "expected": true,
+    "explanation": "The server should have TLS 1.2 enabled. For more information: https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/certificate-overview?view=sql-server-ver16",
```

Comment:
Link related to SQL server on-prem. Might need to change it to https://learn.microsoft.com/en-us/azure/azure-sql/database/connectivity-settings?view=azuresql&tabs=azure-portal#minimal-tls-version